skopt/utils.py

```diff
@@ -388,3 +391,62 @@ def point_aslist(search_space, point_as_dict):
     ]
     return point_as_list
 
+
+def normalize_dimensions(dimensions, base_estimator="GP"):
+    """
+    Normalize all dimensions.
+
+    This is particularly useful for Gaussian process based regressors and is used
+    internally by `gp_minimize.`
+
+    Parameters
+    ----------
+    * `dimensions` [list, shape=(n_dims,)]:
+        List of search space dimensions.
+        Each search dimension can be defined either as
+
+        - a `(upper_bound, lower_bound)` tuple (for `Real` or `Integer`
+          dimensions),
+        - a `(upper_bound, lower_bound, "prior")` tuple (for `Real`
+          dimensions),
+        - as a list of categories (for `Categorical` dimensions), or
+        - an instance of a `Dimension` object (`Real`, `Integer` or
+          `Categorical`).
+
+         NOTE: The upper and lower bounds are inclusive for `Integer`
+         dimensions.
+
+    * `base_estimator` ["GP", "RF", "ET", "GBRT", or sklearn regressor, default="GP"]:
+        optimization estimator to be used.
+        each estimator may use different transformations for dimensions.
+    """
+    if isinstance(base_estimator, str):
+        base_estimator = base_estimator.upper()
+        if base_estimator not in ["GP", "ET", "RF", "GBRT"]:
+            raise ValueError("Valid strings for the base_estimator parameter " +
+                             "are: 'RF', 'ET', 'GBRT', or 'GP'.")
+    elif not is_regressor(base_estimator):
+        raise ValueError("base_estimator has to be a regressor.")
+
+    if base_estimator == "GP" or base_estimator == GaussianProcessRegressor:
```

Comment:
@MechCoder we would then remove the `base_estimator="GP"` argument, the checks for expected strings associated with that argument, and the check for the regressor, right? 

Basically, we would keep lines 432-449?